tracing-appender/src/non_blocking.rs

```diff
@@ -8,14 +8,21 @@ use std::sync::Arc;
 use std::thread::JoinHandle;
 use tracing_subscriber::fmt::MakeWriter;
 
+/// Default number of lines to buffer before dropping logs if `NonBlocking` is is lossy.
 pub const DEFAULT_BUFFERED_LINES_LIMIT: usize = 128_000;
 
+/// `WorkerGuard` is responsible for informing the writer to flush logs in the case of a panic.
+///
+/// This guard should be held for as long as possible and should not be dropped accidentally.
+/// It contains a reference to an `AtomicBool` which is used to inform the off-thread writer's
+/// worker to shutdown and flush logs.
 #[derive(Debug)]
 pub struct WorkerGuard {
     guard: Option<JoinHandle<()>>,
     shutdown_signal: Arc<AtomicBool>,
 }
 
+/// `NonBlocking` is an off-thread writer
```

Comment:
/// A non-blocking writer.
///
/// Writing to an output, such as `std::io::stdout` or a file, is typically a blocking operation.
/// This means that a `Subscriber` where events are logged as they occur will block any threads 
/// emitting events until the events have been logged. This type provides a way to move logging 
/// out of an application's data path by sending log messages to a dedicated worker thread. Any
/// logs written to a `NonBlocking` writer will be forwarded to a corresponding worker to be output.
///
/// This struct implements the [`MakeWriter` trait][make_writer] from the `tracing-subscriber`
/// crate, and can be used with the [`tracing_subscriber::fmt`][fmt] module, or with any other 
/// subscriber implementation that uses the `MakeWriter` interface.
///
/// [make_writer]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/trait.MakeWriter.html
/// [fmt]: https://docs.rs/tracing-subscriber/latest/tracing_subscriber/fmt/index.html